cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -58,18 +59,40 @@ void column_buffer::create(size_type _size,
  */
 std::unique_ptr<column> make_column(column_buffer& buffer,
                                     column_name_info* schema_info,
+                                    std::optional<reader_column_schema> const& schema,
                                     rmm::cuda_stream_view stream,
                                     rmm::mr::device_memory_resource* mr)
 {
   if (schema_info != nullptr) { schema_info->name = buffer.name; }
 
   switch (buffer.type.id()) {
     case type_id::STRING:
-      if (schema_info != nullptr) {
-        schema_info->children.push_back(column_name_info{"offsets"});
-        schema_info->children.push_back(column_name_info{"chars"});
+      if (!schema.value_or(reader_column_schema{}).is_enabled_convert_binary_to_strings()) {
```

Comment:
Complete nitpick: We have an `if (!is_enabled_to_strings) {1} else {2}` construct here. I honestly thought we had a bug, because I missed the `!`.

We might consider unwinding this to `if (is_enabled_to_strings) {2} else {1}`.

Reply:
This is a good idea and helps clear this up.